### PR TITLE
06518 6516 delete platform metrics

### LINKED ISSUE #6518: Get rid of PlatformMetrics.java
The class `PlatformMetrics.java` is thoroughly cursed. Break this class into smaller chunks, and embed these chunks where they logically belong. `SwirldsPlatform.java` shouldn't be exposing lots of getter methods to support metrics.

### PR DESCRIPTION
**Pull request by cody-littley**
Closes #6518

Breaking changes:

- removed the following metrics
  - write: this is just outputting a setting  
  - simCallSyncsMax: this is just outputting a setting
   - TLS: this is just outputting a setting
   - icSync/sec: this is no longer necessary now that we have the metrics family disconnects/sec_with_[NODE-ID]
   - irSync/sec: this is no longer necessary now that we have the metrics family disconnects/sec_with_[NODE-ID]
   - name: not a useful metric

### CODE REVIEW

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/components/state/DefaultStateManagementComponent.java [hunk 1]
```diff
@@ -174,7 +180,7 @@ public DefaultStateManagementComponent(
             @NonNull final PreConsensusEventWriter preConsensusEventWriter,
             @NonNull final Supplier<PlatformStatus> getPlatformStatus) {
 
-        Objects.requireNonNull(context, "context");
+        Objects.requireNonNull(platformContext, "context");
```
Comment (alittley) [inline platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/components/state/DefaultStateManagementComponent.java:183]: Variable name / string mismatch

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/components/state/DefaultStateManagementComponent.java [hunk 2]
```diff
@@ -136,8 +138,12 @@ public class DefaultStateManagementComponent implements StateManagementComponent
 
     private final StateConfig stateConfig;
 
+    private static final RunningAverageMetric.Config AVG_ROUND_SUPERMAJORITY_CONFIG = new RunningAverageMetric.Config(
+                    PLATFORM_CATEGORY, "roundSup")
+            .withDescription("latest round with state signed by a supermajority");
```
Comment (alittley) [inline platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/components/state/DefaultStateManagementComponent.java:143]: Unit could be defined as `round`. Also, is changing this terrible name off the table? (feel free to ignore entirely, if improving these various metrics is outside of scope)

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/components/CriticalQuorumImpl.java [hunk 3]
```diff
@@ -49,35 +54,39 @@ public class CriticalQuorumImpl implements CriticalQuorum {
...
+     * @param addressBook         the source address book
+     * @param considerBothParents true if both parents should be checked for critical quorum and false for just the self
+     *                            parent
+     * @param thresholdSoftening  'soften' the threshold by this many events, the higher the number, the less strict the
+     *                            quorum is
      */
     public CriticalQuorumImpl(
-            final AddressBook addressBook, final boolean considerBothParents, final int thresholdSoftening) {
+            @NonNull final Metrics metrics,
+            @NonNull final long selfId,
```
Comment (alittley) [inline platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/components/CriticalQuorumImpl.java:85]: missing doc for `selfId`

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/components/EventMapper.java [hunk 4]
```diff
@@ -47,19 +52,22 @@ public class EventMapper implements EventAddedObserver, SelfEventStorage, Cleara
     /**
      * Constructor
      *
-     * @param selfId
-     * 		this node's {@link NodeId}
+     * @param selfId this node's {@link NodeId}
      */
-    public EventMapper(final NodeId selfId) {
+    public EventMapper(@NonNull final Metrics metrics, final NodeId selfId) {
```
Comment (alittley) [inline platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/components/EventMapper.java:57]: missing doc for `metrics`

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/eventhandling/EventTransactionPool.java [hunk 5]
```diff
@@ -77,9 +81,19 @@ public EventTransactionPool() {
      * @param inFreeze
      * 		Indicates if the system is currently in a freeze
      */
-    public EventTransactionPool(final SettingsProvider settings, final BooleanSupplier inFreeze) {
+    public EventTransactionPool(
+            @NonNull final Metrics metrics, final SettingsProvider settings, final BooleanSupplier inFreeze) {
```
Comment (alittley) [inline platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/eventhandling/EventTransactionPool.java:85]: Javadoc for `metrics`

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/eventhandling/EventTransactionPool.java [hunk 6]
```diff
@@ -77,9 +81,19 @@ public EventTransactionPool() {
...
         this.settings = settings;
         this.inFreeze = inFreeze;
+
+        metrics.getOrCreate(
+                new FunctionGauge.Config<>(INFO_CATEGORY, "transEvent", Integer.class, this::getTransEventSize)
```
Comment (alittley) [inline platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/eventhandling/EventTransactionPool.java:90]: Is there a reason you inline the metric configs in some places, and have a static member variable in others?

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/gossip/shadowgraph/SimultaneousSyncThrottle.java [hunk 7]
```diff
@@ -37,9 +43,26 @@ public class SimultaneousSyncThrottle {
...
+                    PLATFORM_CATEGORY, "simSyncs")
+            .withDescription("avg number of simultaneous syncs happening at any given time")
+            .withFormat(FORMAT_9_6);
+
+    private static final RunningAverageMetric.Config AVG_SIM_LISTEN_SYNCS_CONFIG = new RunningAverageMetric.Config(
+                    PLATFORM_CATEGORY, "simListenSyncs")
+            .withDescription("avg number of simultaneous listening syncs happening at any given time")
+            .withFormat(FORMAT_9_6);
+
+    public SimultaneousSyncThrottle(@NonNull final Metrics metrics, final int maxListenerSyncs) {
```
Comment (alittley) [inline platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/gossip/shadowgraph/SimultaneousSyncThrottle.java:56]: Missing javadoc for public method

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/state/address/AddressBookMetrics.java [hunk 8]
```diff
@@ -0,0 +1,45 @@
...
+import edu.umd.cs.findbugs.annotations.NonNull;
+
+/**
+ * A utility class to encapsulate the metrics for the address book.
+ */
+public final class AddressBookMetrics {
+
+    private AddressBookMetrics() {}
+
+    public static void registerAddressBookMetrics(
```
Comment (alittley) [inline platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/state/address/AddressBookMetrics.java:34]: Missing javadoc for public method

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/event/EventCounter.java [hunk 9]
```diff
@@ -50,4 +60,9 @@ public static void eventCreated() {
     public static void eventCleared() {
         numEventsInMemory.decrementAndGet();
     }
+
+    public static void registerMetrics(@NonNull final Metrics metrics) {
```
Comment (alittley) [inline platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/event/EventCounter.java:64]: Consider renaming method to match the pattern of other new metric collection classes `registerEventCounterMetrics`

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/components/CriticalQuorumImpl.java [hunk 10]
```diff
@@ -49,35 +54,39 @@ public class CriticalQuorumImpl implements CriticalQuorum {
...
+            final AddressBook addressBook,
```
Comment (edward-swirldslabs) [inline platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/components/CriticalQuorumImpl.java:86]: Nullability of AddressBook here?

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/components/EventMapper.java (hunk 4 above)
Comment (edward-swirldslabs) [reply platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/components/EventMapper.java:57]: And Nullability of NodeId

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/eventhandling/EventTransactionPool.java (hunk 5 above)
Comment (edward-swirldslabs) [reply platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/eventhandling/EventTransactionPool.java:85]: `@Nullable` for other arguments.

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/eventhandling/PreConsensusEventHandler.java [hunk 11]
```diff
@@ -69,19 +76,21 @@ public class PreConsensusEventHandler implements PreConsensusEventObserver, Clea
...
+     * @param consensusMetrics
      * 		metrics relating to consensus
      */
     public PreConsensusEventHandler(
+            @NonNull final Metrics metrics,
             final ThreadManager threadManager,
             final NodeId selfId,
             final SwirldStateManager swirldStateManager,
-            final ConsensusMetrics metrics) {
+            final ConsensusMetrics consensusMetrics) {
```
Comment (edward-swirldslabs) [inline platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/eventhandling/PreConsensusEventHandler.java:87]: Nullability?

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/components/state/DefaultStateManagementComponent.java (hunk 1 above)
Comment (cody-littley) [reply platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/components/state/DefaultStateManagementComponent.java:183]: removed the strings

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/components/state/DefaultStateManagementComponent.java (hunk 2 above)
Comment (cody-littley) [reply platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/components/state/DefaultStateManagementComponent.java:143]: done

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/components/CriticalQuorumImpl.java (hunk 3 above)
Comment (cody-littley) [reply platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/components/CriticalQuorumImpl.java:85]: added

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/components/EventMapper.java (hunk 4 above)
Comment (cody-littley) [reply platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/components/EventMapper.java:57]: fixed, and fixed

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/event/EventCounter.java (hunk 9 above)
[the remark this answers is not in the document]
Comment (cody-littley) [reply platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/event/EventCounter.java:64]: added

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/eventhandling/EventTransactionPool.java [hunk 12]
```diff
@@ -64,9 +69,8 @@ public class EventTransactionPool implements TransactionPool, TransactionSupplie
     protected final SettingsProvider settings;
 
     // Used for creating spy objects for unit tests
-    public EventTransactionPool() {
-        settings = null;
-        inFreeze = null;
+    public EventTransactionPool(@NonNull final Metrics metrics) {
```
[the remark this answers is not in the document]
Comment (cody-littley) [reply platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/eventhandling/EventTransactionPool.java:72]: Removed this constructor.

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/eventhandling/EventTransactionPool.java (hunk 5 above)
Comment (cody-littley) [reply platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/eventhandling/EventTransactionPool.java:85]: Fixed

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/eventhandling/EventTransactionPool.java (hunk 6 above)
Comment (cody-littley) [reply platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/eventhandling/EventTransactionPool.java:90]: Yes. The `FunctionGauge` style metrics require the function to be provided to the configuration object, and you can only obtain the function once the object is constructed.

For other types of metrics, it is possible to statically define the metric's configuration.

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/gossip/shadowgraph/SimultaneousSyncThrottle.java (hunk 7 above)
Comment (cody-littley) [reply platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/gossip/shadowgraph/SimultaneousSyncThrottle.java:56]: added

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/state/address/AddressBookMetrics.java (hunk 8 above)
Comment (cody-littley) [reply platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/state/address/AddressBookMetrics.java:34]: added

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/event/EventCounter.java (hunk 9 above)
Comment (cody-littley) [reply platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/event/EventCounter.java:64]: renamed

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/FreezeMetrics.java [hunk 13]
```diff
@@ -0,0 +1,53 @@
...
+     *
+     * @param metrics                   the metrics engine
+     * @param freezeManager             the freeze manager
+     * @param startUpEventFrozenManager the start up event frozen manager
+     */
+    public static void registerFreezeMetrics(
+            @NonNull final Metrics metrics,
+            @NonNull final FreezeManager freezeManager,
+            @NonNull final StartUpEventFrozenManager startUpEventFrozenManager) {
+
```
[the remark this answers is not in the document]
Comment (cody-littley) [reply platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/FreezeMetrics.java:43]: added

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/components/CriticalQuorumImpl.java (hunk 10 above)
Comment (cody-littley) [reply platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/components/CriticalQuorumImpl.java:86]: fixed

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/components/CriticalQuorumImpl.java [hunk 14]
```diff
@@ -86,16 +95,25 @@ public CriticalQuorumImpl(
...
      *
-     * @param addressBook
-     * 		the source address book
+     * @param metrics     the metrics engine
+     * @param selfId      the id of this node
+     * @param addressBook the source address book
      */
-    public CriticalQuorumImpl(final AddressBook addressBook) {
-        this(addressBook, DEFAULT_BOTH_PARENTS, DEFAULT_THRESHOLD_SOFTENING);
+    public CriticalQuorumImpl(@NonNull final Metrics metrics, final long selfId, final AddressBook addressBook) {
```
[the remark this answers is not in the document]
Comment (cody-littley) [reply platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/components/CriticalQuorumImpl.java:115]: fixed

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/eventhandling/PreConsensusEventHandler.java (hunk 11 above)
Comment (cody-littley) [reply platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/eventhandling/PreConsensusEventHandler.java:87]: fixed

**Review by alittley — APPROVED** [review]

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/eventhandling/EventTransactionPool.java [hunk 15]
```diff
@@ -63,30 +68,34 @@ public class EventTransactionPool implements TransactionPool, TransactionSupplie
...
      */
-    public EventTransactionPool(final SettingsProvider settings, final BooleanSupplier inFreeze) {
+    public EventTransactionPool(
+            @NonNull final Metrics metrics,
+            @Nullable SettingsProvider settings,
+            @Nullable final BooleanSupplier inFreeze) {
         this.settings = settings;
         this.inFreeze = inFreeze;
+
+        metrics.getOrCreate(
```
Comment (hendrikebbers) [inline platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/eventhandling/EventTransactionPool.java:85]: Null check

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/gossip/shadowgraph/SimultaneousSyncThrottle.java [hunk 16]
```diff
@@ -37,21 +42,42 @@ public class SimultaneousSyncThrottle {
...
+     * Construct a new SimultaneousSyncThrottle
+     *
+     * @param metrics          the metrics engine
+     * @param maxListenerSyncs the maximum number of listener syncs that can happen at the same time
+     */
+    public SimultaneousSyncThrottle(@NonNull final Metrics metrics, final int maxListenerSyncs) {
         this.maxListenerSyncs = maxListenerSyncs;
         simSyncThrottleLock = new ConcurrentHashMap<>();
+
+        final RunningAverageMetric avgSimSyncs = metrics.getOrCreate(AVG_SIM_SYNCS_CONFIG);
```
Comment (hendrikebbers) [inline platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/gossip/shadowgraph/SimultaneousSyncThrottle.java:65]: null check

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/state/address/AddressBookMetrics.java [hunk 17]
```diff
@@ -0,0 +1,52 @@
...
+     * Register the metrics for the address book.
+     *
+     * @param metrics     the metrics engine
+     * @param addressBook the address book to register metrics for
+     * @param selfId      the ID of the node
+     */
+    public static void registerAddressBookMetrics(
+            @NonNull final Metrics metrics, @NonNull final AddressBook addressBook, @NonNull final NodeId selfId) {
+
+        metrics.getOrCreate(new FunctionGauge.Config<>(INFO_CATEGORY, "memberID", Long.class, selfId::getId)
```
Comment (hendrikebbers) [inline platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/state/address/AddressBookMetrics.java:44]: null checks

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/eventhandling/EventTransactionPool.java (hunk 15 above)
Comment (cody-littley) [reply platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/eventhandling/EventTransactionPool.java:85]: This is dereferenced in the constructor and will throw an exception promptly if null.

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/gossip/shadowgraph/SimultaneousSyncThrottle.java (hunk 16 above)
Comment (cody-littley) [reply platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/gossip/shadowgraph/SimultaneousSyncThrottle.java:65]: This is dereferenced in the constructor and will throw an exception promptly if null.

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/state/address/AddressBookMetrics.java (hunk 17 above)
Comment (cody-littley) [reply platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/state/address/AddressBookMetrics.java:44]: All parameters are dereferenced in the constructor and will throw an exception promptly if null. Lol, and some of them did throw in unit tests that I had to fix. 🙃

**Pull request merged**